### PR TITLE
TLS 1.3: Key Generation: change some key generation functions to static

### PR DESCRIPTION
**Pull request by yanrayw**
## Description

Fix: #6675 

## Gatekeeper checklist

- [x] **changelog** not required, part of enhancing key generation in TLS 1.3.
- [x] **backport** not required
- [x] **tests** not required

## Comments
This PR only fixes one issue listed in https://github.com/Mbed-TLS/mbedtls/issues/6675.

### CODE REVIEW

File: library/ssl_tls13_keys.c [hunk 1]
```diff
@@ -629,7 +638,25 @@ int mbedtls_ssl_tls13_derive_resumption_master_secret(
...
+ *        can be used to derive the handshake traffic keys.
+ *
+ * \param ssl  The SSL context to operate on. This must be in key schedule
+ *             stage \c Handshake.
+ *
+ * \returns    \c 0 on success.
+ * \returns    A negative error code on failure.
+ */
+MBEDTLS_CHECK_RETURN_CRITICAL
+static int mbedtls_ssl_tls13_key_schedule_stage_application(
```
Comment (yuhaoth) [inline library/ssl_tls13_keys.c:658]:
```suggestion
static int ssl_tls13_key_schedule_stage_application(
```

File: library/ssl_tls13_keys.c [hunk 2]
```diff
@@ -1295,10 +1337,26 @@ int mbedtls_ssl_tls13_key_schedule_stage_early( mbedtls_ssl_context *ssl )
...
+ *             key schedule stage \c Handshake, see
+ *             mbedtls_ssl_tls13_key_schedule_stage_handshake().
+ * \param traffic_keys The address at which to store the handshake traffic key
+ *                     keys. This must be writable but may be uninitialized.
+ *
+ * \returns    \c 0 on success.
+ * \returns    A negative error code on failure.
+ */
+MBEDTLS_CHECK_RETURN_CRITICAL
+static int mbedtls_ssl_tls13_generate_handshake_keys(
```
Comment (yuhaoth) [inline library/ssl_tls13_keys.c:1357]:
```suggestion
static int ssl_tls13_generate_handshake_keys(
```

File: library/ssl_tls13_keys.c [hunk 3]
```diff
@@ -1503,10 +1585,24 @@ int mbedtls_ssl_tls13_key_schedule_stage_handshake( mbedtls_ssl_context *ssl )
...
+ *             mbedtls_ssl_tls13_key_schedule_stage_application().
+ * \param traffic_keys The address at which to store the application traffic key
+ *                     keys. This must be writable but may be uninitialized.
+ *
+ * \returns    \c 0 on success.
+ * \returns    A negative error code on failure.
  */
-int mbedtls_ssl_tls13_generate_application_keys(
+MBEDTLS_CHECK_RETURN_CRITICAL
+static int mbedtls_ssl_tls13_generate_application_keys(
```
Comment (yuhaoth) [inline library/ssl_tls13_keys.c:1605]:
```suggestion
static int ssl_tls13_generate_application_keys(
```

**Review by yuhaoth — CHANGES_REQUESTED** [review]
Some nitpick improvements. And others look good to me

**Comment by yanrayw** [comment]
Since new code style was introduced, this PR was rebased to resolve all the conflicts. In addition, this PR was simplified from 4 commits to 2 commits since we don't have to focus on code style anymore.

**Review by yuhaoth — DISMISSED** [review]
LGTM

**Review by xkqian — CHANGES_REQUESTED** [review]

**Review by xkqian — APPROVED** [review]
LGTM

**Review by yuhaoth — APPROVED** [review]
LGTM

**Pull request merged**